### PR TITLE
fix(KTO-441): option to display the ID instead of the description as …

### PR DESCRIPTION
**Pull request by tplevko**
…step label

fixes https://issues.redhat.com/browse/KTO-441

The option was added to the settings page:

Screenshot from 2024-07-01 09-52-10

The ID is used either when selected, or in case the "description" is selected, but not yet specified in the node form.

### CODE REVIEW

**Review by lordrip — COMMENTED** [review]
Thanks @tplevko for taking care of this, unfortunately, in order to make it work in VSCode, we need to provide a settings adapter that handles VSCode settings as well. Is it ok for you to hold this PR for a bit until we create that first?

**Comment by tplevko** [comment]
> Thanks @tplevko for taking care of this, unfortunately, in order to make it work in VSCode, we need to provide a settings adapter that handles VSCode settings as well. Is it ok for you to hold this PR for a bit until we create that first?

sure, let's wait for the VSCode settings adapter

**Comment by lordrip** [comment]
@tplevko, after merging https://github.com/KaotoIO/kaoto/pull/1264, we would need to introduce an intermediate PR just adding the new `nodeLabel` setting without using it, this way we can merge that one and then merge the VSCode part from it.

**Comment by lordrip** [comment]
@tplevko, I think we can rebase this PR and merge it, as VSCode is in a middle ground and we still have things to polish before making it available to the public.

File: packages/ui/src/models/settings/settings.model.ts
Comment (lordrip) [inline packages/ui/src/models/settings/settings.model.ts:13]: I think this should be an `enum`, we can take care of it at another time.

**Review by lordrip — APPROVED** [review]

**Pull request merged**
